EIPS/eip-zodiac.md

```diff
@@ -0,0 +1,175 @@
+
+    /// @dev Returns array of modules.
+    /// @param start Start of the page.
+    /// @param pageSize Maximum number of modules that should be returned.
+    /// @return array Array of modules.
+    /// @return next Start of the next page.
+    function getModulesPaginated(address start, uint256 pageSize)
+        external
+        view
+        returns (address[] memory array, address next);
+}
+```
+
+```solidity
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+interface IGuard {
+    function checkTransaction(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes memory signatures,
+        address msgSender
+    ) external;
+
+    function checkAfterExecution(bytes32 txHash, bool success) external;
+}
+
+```
+
+```solidity
+pragma solidity >=0.7.0 <0.9.0;
+
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "./BaseGuard.sol";
+
+/// @title Guardable - A contract that manages fallback calls made to this contract
+contract Guardable is OwnableUpgradeable {
+    address public guard;
+
+    event ChangedGuard(address guard);
+
+    /// `guard_` does not implement IERC165.
+    error NotIERC165Compliant(address guard_);
+
+    /// @dev Set a guard that checks transactions before execution.
+    /// @param _guard The address of the guard to be used or the 0 address to disable the guard.
+    function setGuard(address _guard) external onlyOwner {
+        if (_guard != address(0)) {
+            if (!BaseGuard(_guard).supportsInterface(type(IGuard).interfaceId))
+                revert NotIERC165Compliant(_guard);
+        }
+        guard = _guard;
+        emit ChangedGuard(guard);
+    }
+
+    function getGuard() external view returns (address _guard) {
+        return guard;
+    }
+}
+```
+
+## Rationale
+In designing the Zodiac standard, we inevitably needed to define a standard interface for modular programmable accounts (avatars). Rather than writing this from scratch, we elected to use the existing interfaces from the [Gnosis Safe's Module Manager](https://github.com/gnosis/safe-contracts/blob/main/contracts/base/ModuleManager.sol) and from the [Gnosis Safe's Guard Manager](https://github.com/gnosis/safe-contracts/blob/main/contracts/base/GuardManager.sol), since the Gnosis Safe is by far the most widely used programmable account for DAOs, other organizations, and individuals.
+
+## Backwards Compatibility
+No backward compatibility issues are introduced by this standard.
+
+## Reference Implementation
+A reference implementation can be found in the [gnosis/zodiac](https://github.com/gnosis/zodiac) repository.
```

Comment:
Reference Implementation should be in this document or  directory `/EIPs/assets/eip-5005/`

Reply:
Removing the reference implementation, since it's not really critical to the EIP.